doc/src/documentation_style_guide.rst

```diff
@@ -0,0 +1,1019 @@
+a math context because it does not render as the Greek letter (α).
+
+Style Preferences
+-----------------
+
+**Spelling and Punctuation**
+
+All narrative writing in SymPy follows American spelling and punctuation
+standards. For example, “color” is preferred over “colour” and commas should be
+placed inside of quotation marks.
+
+**Examples**
+
+::
+
+  If the ``line_color`` aesthetic is a function of arity 1, then the coloring is
+  a function of the x value of a point.
+
+  The term "unrestricted necklace," or "bracelet," is used to imply an object
+  that can be turned over or a sequence that can be reversed.
+
+If there is any ambiguity about the spelling of a word, such as in the case of
+a function named after a person, refer to the spelling of the actual SymPy
+function.
+
+For example, Chebyshev polynomials are named after Pafnuty Lvovich Tchebychev,
+whose name is sometimes transliterated from Russian to be spelled with a “T,”
+but in SymPy it should always be spelled “Chebyshev” to refer to the SymPy
+function.
+
+**Example**
+
+::
+
+  class chebyshevt(OrthogonalPolynomial):
+      r"""
+      Chebyshev polynomial of the first kind, $T_n(x)$
+      ...
+
+      """
+
+**Capitalization**
+
+Title case capitalization is preferred in all SymPy headings.
+
+**Example**
+
+::
+
+  What is Symbolic Computation?
+  -----------------------------
+
+**Grammar**
+
+WORK IN PROGRESS: Google Season of Docs technical writer will track common
+grammar mistakes (e.g., using certain articles correctly) while editing and add
+to this guide.
+
+Tone Preferences
+----------------
+
+Across SymPy documentation, please write in:
+
+* The present tense (e.g., In the following section, we are going to learn...)
+* The first-person inclusive plural (e.g., We did this the long way, but now we
+  can try it the short way...)
+* Use the generic pronoun “you” instead of “one.” Or use “the reader” or “the
+  user.” (e.g., You can access this function by... The user can then access
+  this function by...)
+* Use the gender-neutral pronoun “they” instead of “he” or “she.” (e.g., A good
+  docstring tells the user exactly what they need to know.)
+
+Avoid extraneous or belittling words such as “obviously,” “easily,” “simply,”
+“just,” or “straightforward.”
+
+Avoid unwelcoming or judgement-based phrases like “That is wrong.” Instead use
+friendly and inclusive language like “A common mistake is...”
+
+Avoid extraneous phrases like, “we just have to do one more thing.”
+
+Docstring Guidelines
+====================
+
+To contribute to SymPy’s docstrings, please read these guidelines in full.
+
+A documentation string (docstring) is a string literal that occurs as the first
+statement in a module, function, class, or method definition. Such a docstring
+becomes the ``__doc__`` special attribute of that object.
+
+**Example**
+
+Here is a basic docstring::
+
+  def fdiff(self, argindex=1):
+      """
+      Returns the first derivative of a Heaviside Function.
```

Comment:
Is this supposed to be compatible with http://www.pydocstyle.org/?

Because it's not:

    in public function `fdiff`:
        D401: First line should be in imperative mood ('Return', not 'Returns')

Reply:
@lglattly can clarify, but I think we decided that we aren't following some of the rules from the PEP, including that one. The reasoning being that we already don't do this, and it's pretty common for other projects to not do it either. This is also why we don't link to the PEP from this document.

Reply:
pydocstyle seems to follwo PEP 257, but that pep doesn't seem accepted. Its status is active, since 2001 when it was created. If PEP 257 is an accepted PEP, then it might be good to follow its specs.

Reply:
Actually, how do you know if a PEP is accepted?

Reply:
It has status "active". See https://www.python.org/dev/peps/pep-0001/#pep-review-resolution, particularly the last paragraph of that section. 

I think that PEP is more or less "accepted", but as I said, almost no project actually follows this rule (and IMO rightly).

Reply:
There seem to be a lot of different PEP statuses, see e.g. those categories: https://www.python.org/dev/peps/

PEP 257 hast the same "Active" status as [PEP 8](https://www.python.org/dev/peps/pep-0008/) and [PEP 20 (The Zen of Python)](https://www.python.org/dev/peps/pep-0020/).

I'm not an expert, but I guess an "Active" PEP is basically the law.

I guess the "bad" statuses are: Deferred, Withdrawn, Rejected, Superseded.

I think "Accepted" actually has a different meaning: It may not yet be implemented.
Once it is implemented, the status is changed to "Final". So "Accepted" is just a temporary status.

I don't know what the difference between "Final" and "Active" is.

Reply:
We chose not to link to PEP 257 because there were a few guidelines in there that SymPy doesn't follow and we thought linking to it might cause confusion. The above example is copy and pasted out of SymPy's current documentation, where both "Return" and "Returns" are used (Returns even more frequently) and so both are accepted, along with other variations on the single-sentence summary, such as those that are more like noun definitions. More info on writing the single-sentence summary is below in its section.